neo/Consensus/ConsensusService.cs

```diff
@@ -122,7 +119,7 @@ private void CheckPreparations()
                 ConsensusPayload payload = context.MakeCommit();
                 Log($"send commit");
                 context.State |= ConsensusState.CommitSent;
-                store.Put(ContextSerializationPrefix, new byte[0], context.ToArray());
+                context.WriteContextToStore(store);
```

Comment:
Lets add this after relaying, we could gain some network propagation time.

Reply:
I think we are currently under the thinking it isn't worth slowing down the common case where no recovery is needed, to warrant persisting everything more than once. If we do decide to do it, persistence needs to be done in a way that is more optimal when saving a second time when commit occurs (doesn't need to save the transactions again).

Reply:
It SHOULD write recovery logs before relaying. Otherwise, if the node crashes between relaying and writing, the commit state cannot be locked.

Reply:
I'll switch it back. I though this is why it was in this order. However; are we sure that LevelDB actually writes it to disk synchronously?

Reply:
There is an option to make it synchronously.

Reply:
https://github.com/neo-project/neo/blob/6ae3c6d8324663ac77d747a0e71b03a0617b6ece/neo/IO/Data/LevelDB/WriteOptions.cs#L10-L16

Reply:
Actually, I didn't change this. So there is nothing I need to do here. Just resolve this conversation is all.

Reply:
I will wait for @vncoelho to reply though.

Reply:
> However; are we sure that LevelDB actually writes it to disk synchronously?

So, I have not considered this issue before, we should change it to be synchronous.

Reply:
I included changes to make it synchronous.